packages/rum-core/src/common/apm-server.js

```diff
@@ -51,29 +51,32 @@ class ApmServer {
       return
     }
     this.initialized = true
-
-    this.initErrorQueue()
-    this.initTransactionQueue()
+    this._initQueue()
   }
 
-  createMetaData() {
-    const cfg = this._configService
-    const metadata = {
-      service: {
-        name: cfg.get('serviceName'),
-        version: cfg.get('serviceVersion'),
-        agent: {
-          name: 'js-base',
-          version: cfg.version
-        },
-        language: {
-          name: 'javascript'
-        },
-        environment: cfg.get('environment')
-      },
-      labels: cfg.get('context.tags')
+  _initQueue() {
+    const queueLimit = this._configService.get('queueLimit')
+    const flushInterval = this._configService.get('flushInterval')
+    const limit = this._configService.get('eventsPerMin')
```

Comment:
Do you we need to check if any of these config options are undefined?

Reply:
Hmmm, We can be defensive and handle all cases. I am not 100% sold on the idea of testing user config when we clearly mention each config in the docs and explain the type as well.

Reply:
I think some validations would be useful at least when the user sets these values would be useful, but should do that separately.

Reply:
Yeah, we need to have validations for all configs and report warnings for non important ones and keep it as error for others like we have for `serverUrl`